plugins/woocommerce/tests/php/src/Internal/Traits/AccessiblePrivateMethodsTest.php

```diff
@@ -0,0 +1,439 @@
+	 * @testdox New add_(static_)action and add_(static_)filter methods can be used to register private and protected class methods as hook callbacks.
+	 *
+	 * @param bool $use_string_syntax True to set hooks passing the method name, false to use the standard [$this, 'method_name'] syntax.
+	 */
+	public function test_private_and_protected_hook_handler_methods_can_be_made_accessible( bool $use_string_syntax ) {
+		//phpcs:disable Squiz.Commenting
+		$sut = new class($use_string_syntax) {
+			use AccessiblePrivateMethods;
+
+			public $action_argument = null;
+
+			public static $static_action_argument = null;
+
+			public function __construct( bool $use_string_syntax ) {
+				if ( $use_string_syntax ) {
+					$this->add_action( 'action_handled_privately', 'handle_action' );
+					$this->add_action( 'filter_handled_privately', 'handle_filter' );
+				} else {
+					$this->add_action( 'action_handled_privately', array( $this, 'handle_action' ) );
+					$this->add_action( 'filter_handled_privately', array( $this, 'handle_filter' ) );
+				}
+			}
+
+			//phpcs:ignore WooCommerce.Functions.InternalInjectionMethod.MissingInternalTag
+			final public static function init( bool $use_string_syntax ) {
+				if ( $use_string_syntax ) {
+					self::add_static_action( 'static_action_handled_privately', 'handle_static_action' );
+					self::add_static_action( 'static_filter_handled_privately', 'handle_static_filter' );
+				} else {
+					self::add_static_action( 'static_action_handled_privately', array( __CLASS__, 'handle_static_action' ) );
+					self::add_static_action( 'static_filter_handled_privately', array( __CLASS__, 'handle_static_filter' ) );
+				}
+			}
+
+			private function handle_action( $argument ) {
+				$this->action_argument = $argument;
+			}
+
+			private function handle_filter( $argument ) {
+				return 'Filter argument: ' . $argument;
+			}
+
+			private static function handle_static_action( $argument ) {
+				self::$static_action_argument = $argument;
+			}
+
+			private static function handle_static_filter( $argument ) {
+				return 'Static filter argument: ' . $argument;
+			}
+		};
+		//phpcs:enable Squiz.Commenting
+
+		$sut::init( $use_string_syntax );
+
+		//phpcs:disable WooCommerce.Commenting.CommentHooks.MissingHookComment
+
+		do_action( 'action_handled_privately', 'foo' );
+		$this->assertEquals( 'foo', $sut->action_argument );
+
+		$filter_result = apply_filters( 'filter_handled_privately', 'bar' );
+		$this->assertEquals( 'Filter argument: bar', $filter_result );
+
+		do_action( 'static_action_handled_privately', 'fizz' );
+		$this->assertEquals( 'fizz', $sut::$static_action_argument );
+
+		$filter_result = apply_filters( 'static_filter_handled_privately', 'buzz' );
+		$this->assertEquals( 'Static filter argument: buzz', $filter_result );
+
+		//phpcs:enable WooCommerce.Commenting.CommentHooks.MissingHookComment
+	}
+
+	/**
+	 * @testdox Trying to make a public method accessible with mark_method_as_accessible does nothing.
```

Comment:
"Does nothing" is perhaps a wrong description, what I mean is that for practical purposes nothing changes, since `__call(Static)` iss never called for public methods.